Makefile

```diff
@@ -2147,6 +2151,15 @@ GIT-USER-AGENT: FORCE
 		echo '$(GIT_USER_AGENT_SQ)' >GIT-USER-AGENT; \
 	fi
 
+GIT_BUILT_FROM_COMMIT = $(eval GIT_BUILT_FROM_COMMIT := $$(shell \
```

Comment:
Do we need a `?=` here, to avoid this `eval` from happening in case the user specified it on the command-line? I.e. like this:

```
GIT_BUILT_FROM_COMMIT ?= $(shell GIT_CEILING_DIRECTORIES="$(CURDIR)/.." \
    git rev-parse -q --verify HEAD 2>/dev/null)
```

Reply:
The command line-specified variable will override any `=` or `:=` assignment for that variable ([documentation](https://www.gnu.org/software/make/manual/html_node/Overriding.html)). I can't infer any "rule" or pattern for when `?=` vs. `=` is used in this `Makefile`, but the variables most similar to `GIT_BUILT_FROM_COMMIT` (e.g. `GIT_USER_AGENT`) use non-conditional assignments.

Reply:
Okay, thank you! I thought `?=` was the thing to use if one wanted to define some variable that had not yet been defined, but that was not based on reading the documentation 😁